### PR TITLE
tickets/DM-44475: Mount campaign management web app as a sub application to cm-service FastAPI app

### PR DESCRIPTION
**Pull request by emanehab99**
This PR has the following changes:
- Mounting the web app as a sub application to CM FastAPI app
- Add endpoints and templates to home page, campaign steps and step details pages

To run the app, go through the same steps to run cm-service FastAPI app. Home page URL is http://0.0.0.0:8080/web_app/campaigns/

### CODE REVIEW

**Comment by emanehab99** [comment]
> @fritzm I'm not sure why the checks are failing. Any suggestion to make them pass?

Nevermind. I thing I messed up the dependencies when I rebased my branch. The original issue is now fixed. I run into other issues though while I'll fix soon

**Comment by emanehab99** [comment]
@fritzm @eacharles All checks are passing now. Can you please review the PR?

**Comment by eacharles** [comment]
Thanks for all the work.   It looks good.   My only concern duplication of functionality between the WebApp and the routers.   I'm happy to defer that for later, but I'd like @fritzm to chime in.

File: src/lsst/cmservice/web_app/app.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+from sqlalchemy.ext.asyncio import async_scoped_session
+from safir.logging import configure_logging, configure_uvicorn_logging
+
+from lsst.cmservice.config import config
+from lsst.cmservice.web_app.pages.campaigns import search_campaigns, get_campaign_details
+from lsst.cmservice.web_app.pages.steps import get_campaign_steps, get_step_details
+from lsst.cmservice.web_app.pages.step_details import get_step_details_by_id
+
+
+configure_logging(profile=config.profile, log_level=config.log_level, name=config.logger_name)
```
Comment (fritzm) [inline src/lsst/cmservice/web_app/app.py:25]: Are these global inits needed/desireable here, since already down in `main.py`?

**Review by fritzm — APPROVED** [review]
Great -- thanks for this!

I'm thinking a little about whether we want this as a "sub-application" (app.mount)...  Really, its not completely independent; it is sharing database, redis, etc.  Slightly worried that init'ing, final'ing these multiple times might have unforseen ramifications within the safir framework. 🤔

Let's go ahead and (rebase and) merge this as a first step, then contemplate how/where we want to integrate the web app handlers as a subsequent improvement step.

File: src/lsst/cmservice/web_app/app.py (hunk 1 above)
Comment (emanehab99) [reply src/lsst/cmservice/web_app/app.py:25]: I'll double check they are needed here and fix it if needed in the next PR

**Pull request merged**